### DISCUSSION TITLE
Proxy for third parties

### DISCUSSION THREAD

**Discussion by SkinyMonkey** [linkerd/linkerd2#5971] [Ideas] [answered]
Hi!

Is there a way to add, manually or not, proxies in front of third parties/external APIs?

Example : I'm communicating with the Azure API and i'd like to consider it as a "member" of our architecture as it is crucial to our product.

Being able to do this would allow me to:
- monitor/trace the traffic
- limit it
- reproduce some communication problems with the API.

Would this be a new feature or is this already possible with linkerd and if it is possible how could i do that?

Thank you!

**Comment by mateiidavid** [comment, marked as answer]
Hey 👋 can you have a look at [this repository](https://github.com/grampelberg/k8s-egress) and let me know if that works for you? It's a POC on how to do egress with linkerd.

The TL;DR here is that to reach a 3rd party API service, you can inject a normal nginx proxy (which will do the call for you); the linkerd2 proxy should report metrics and give you mTLS on at least some portion of the traffic. If you configure a ServiceProfile I think you can also have some retries and timeouts baked in.

**Reply by SkinyMonkey** [reply]
Nice, i'll check it :)
